daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Utf8.scala

```
@@ -73,10 +73,14 @@ object Utf8 {
   def unpack(s: String): ImmArray[Long] =
     ImmArray(s.codePoints().iterator().asScala.map(_.toLong).iterator.to(Iterable))
 
-  @throws[IllegalArgumentException]
-  def pack(codePoints: ImmArray[Long]): String = {
+  // Converts the List of Unicode code point into a String if all code point are valid.
+  // Returns the first invalid code point as Left otherwise.
+  def pack(codePoints: ImmArray[Long]): Either[Long, String] = {
```

Just better to return the error explicit